src/pybullet_blocks/envs/base_env.py

```diff
@@ -486,6 +486,13 @@ def step(
             if i == 0:
                 for _ in range(self._num_sim_steps_per_step):
                     p.stepSimulation(physicsClientId=self.physics_client_id)
+                    if check_body_collisions(
```

Comment:
this looks good for now, but if we find later that speed is an issue, we may want to move this outside so that we're not checking at every simulations step